server/stream.go

```diff
@@ -1006,16 +1006,6 @@ func (s *Server) checkStreamCfg(config *StreamConfig, acc *Account) (StreamConfi
 		if len(cfg.Sources) > 0 {
 			return StreamConfig{}, NewJSMirrorWithSourcesError()
 		}
-		if cfg.Mirror.FilterSubject != _EMPTY_ {
-			return StreamConfig{}, NewJSMirrorWithSubjectFiltersError()
-		}
-		if cfg.Mirror.OptStartSeq > 0 && cfg.Mirror.OptStartTime != nil {
-			return StreamConfig{}, NewJSMirrorWithStartSeqAndTimeError()
-		}
-		if cfg.Duplicates != time.Duration(0) {
```

Comment:
Why removing this? (or check for startseq/starttime, although those I am not sure if it makes sense or not)

Reply:
This should also be allowed. I only want to mirror last hour, or last 1k msgs.

Reply:
Some chat on slack, but these are breaking changes that broke working code.  Now we support filters (didnt before) and I think dupe check is usable as long as we do the dupe check on the deltas of the original message timestamps

Reply:
DeDupe imo does not need to look at original timestamp, its reference frame is when it receives a message at time T it will wait for de-dupe time window to see if another message arrives with same MsgId. Of course only stores in presence of MsgId to start with..

Reply:
during initial sync - or just whenever there's some lag - you wll get messages arriving in mirror stream at a rate not aligned with the rate at which they arrived in source.

Presumbly the dupe window is time between mesages, and since the rate of delivery that matters there is the source rate I think source timestamp is needed?

Reply:
@kozlovic ah - but i am sure we can handle that better than denying it.  The annoying thing here is that there's a default of 2m, clients might set that i am not sure of them all, but if they do now users must remember to specifically set it to 0 when creating a mirror which is not nice at all.

So mainly looking to a) make this not be annoying b) not be a breaking change as it used to work.

Specifically tools that create a config and later compare it - will now potentially see changes since previously they could set a dedup window that then later vanished and then later resulted in error.

Reply:
@ripienaar I don't know the background, just that there had been issues and a change in server that once deployed would panic, etc.. I did contribute by adding the test, etc.. while working with Matthias on that. Will have to wait for @matthiashanel to comment.

Reply:
ok, so we believe that mirrors should not do de-dupe, that is fine. I can revert, and if its also default just wipe but don't error yes?

Reply:
Meaning, unless there is another reason to not have dedupe window on a mirror, such as which timestamp to consider, this check can go.

Reply:
Well I am not saying you must add that feature. I am saying if you want to do dedupe that’s all that make sense. 

I domy think it’s a priority support. I just want a nicer failure mode.

Reply:
ok reverted last CL.